pkg/controller/snapshotter/snapshotter.go

```diff
@@ -160,78 +166,70 @@ func (vs *volumeSnapshotter) getSnapshotDataFromSnapshotName(snapshotName string
-	var snapshotDataObj crdv1.VolumeSnapshotData
-	snapshotDataName := snapshot.Spec.SnapshotDataName
-	glog.Infof("In waitForPluginSnapshot: %s", snapshotName)
+func (vs *volumeSnapshotter) waitForSnapshot(snapshotName string, snapshot *crdv1.VolumeSnapshot, snapshotDataObj *crdv1.VolumeSnapshotData) error {
+	// Get a fresh VolumeSnapshot from API
+	var snapshotObj crdv1.VolumeSnapshot
 	err := vs.restClient.Get().
-		Name(snapshotDataName).
-		Resource(crdv1.VolumeSnapshotDataResourcePlural).
-		Namespace(v1.NamespaceDefault).
-		Do().Into(&snapshotDataObj)
-	if err != nil {
-		return err
-	}
-
-	pv, err := vs.getPVFromVolumeSnapshot(snapshotName, snapshot)
+		Name(snapshot.Metadata.Name).
+		Resource(crdv1.VolumeSnapshotResourcePlural).
+		Namespace(snapshot.Metadata.Namespace).
+		Do().Into(&snapshotObj)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error getting snapshot object %s from API server: %v", snapshotName, err)
 	}
-	spec := &pv.Spec
-	volumeType := crdv1.GetSupportedVolumeFromPVSpec(spec)
-	if len(volumeType) == 0 {
-		return fmt.Errorf("unsupported volume type found in PV %#v", spec)
+	snapshotDataName := snapshotObj.Spec.SnapshotDataName
+	glog.Infof("In waitForSnapshot: snapshot %s snapshot data %s", snapshotName, snapshotDataName)
+	if snapshotDataObj == nil {
+		return fmt.Errorf("Failed to update VolumeSnapshot for snapshot %s: no VolumeSnapshotData.", snapshotName)
 	}
-	plugin, ok := (*vs.volumePlugins)[volumeType]
-	if !ok {
-		return fmt.Errorf("%s is not supported volume for %#v", volumeType, spec)
+
+	plugin, err := vs.getPlugin(snapshotName, snapshot)
+	if plugin == nil {
+		return fmt.Errorf("Failed to get volume plugin to wait for snapshot creation %s. %v", snapshotName, err)
 	}
 
-	var completed bool = false
+	var newSnapshot *crdv1.VolumeSnapshot = nil
 	var conditions *[]crdv1.VolumeSnapshotCondition = nil
+	var status, newstatus string = "", ""
 	// Wait until the snapshot is successfully created by the plugin or an error occurs that
 	// fails the snapshot creation.
 	for {
-		// TODO(xyang): Use status returned from DescribeSnapshot to update snapshot
-		conditions, completed, err = plugin.DescribeSnapshot(&snapshotDataObj)
-		if !completed {
-			glog.Infof("Snapshot %s creation is not complete yet. Status: [%#v] Retrying...", snapshotName, conditions)
-			if err != nil {
-				// Don't bail out here. The error message could be "current snapshot status is: creating".
-				glog.Infof("Message received when creating snapshot %s: %v", snapshotName, err)
+		conditions, _, err = (*plugin).DescribeSnapshot(snapshotDataObj)
+		if err != nil {
+			glog.Warningf("failed to get snapshot %v, err: %v", snapshotName, err)
+			continue
+		}
+		newstatus = vs.getSimplifiedSnapshotStatus(conditions)
+		if newstatus == statusPending {
+			if newstatus != status {
+				status = newstatus
+				glog.V(5).Info("Snapshot %s creation is not complete yet. Status: [%#v] Retrying...", snapshotName, conditions)
```

Comment:
Typo: Should be Infof not Info...

Reply:
I used "glog.V(5).Info" because there are other places in this repo that already use it so I thought that is the correct usage.  I also found both glog.V(x).Info and glog.V(x).Infof are used in the kubernetes master branch.  Then I see both are correct format in the following glog doc:

https://godoc.org/github.com/golang/glog

I can certainly change it to V(5).Infof.

Reply:
The only reason I noticed this is because I got this in my logs... (Info doesn't recognize the formatting sequences and simply prints all of the arguments out):
I0906 07:33:18.230918    2175 snapshotter.go:206] Snapshot %s creation is not complete yet. Status: [%#v] Retrying...default/snapshot-demo&[{Pending Unknown 2017-09-06 07:33:18.230902742 +0000 UTC m=+1373.230023156  Snapshot is being created}]

It's not really that important. :)

Reply:
Got it:).  Infof is the right format here then.  I'll change it.